internal/util/rocksmq/server/rocksmq/rocksmq_impl.go

```diff
@@ -299,9 +307,14 @@ func (rmq *rocksmq) DestroyTopic(topicName string) error {
 	}
 
 	topicMu.Delete(topicName)
-	rmq.retentionInfo.ackedInfo.Delete(topicName)
-	rmq.retentionInfo.lastRetentionTime.Delete(topicName)
-	rmq.retentionInfo.pageInfo.Delete(topicName)
+	for i, name := range rmq.retentionInfo.topics {
```

Comment:
delete logic is wrong. 
In this for loop, you will skip the check of the next element at each hit.